image/registry_handler.go

```diff
@@ -0,0 +1,94 @@
+package image
+
+import (
+	"strings"
+
+	"github.com/buildpacks/imgutil/remote"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle/cmd"
+)
+
+// RegistryHandler takes care of the registry settings and checks
+//
+//go:generate mockgen -package testmock -destination testmock/registry_handler.go github.com/buildpacks/lifecycle RegistryHandler
+type RegistryHandler interface {
+	EnsureReadAccess(imageRefs ...string) error
+	EnsureWriteAccess(imageRefs ...string) error
+}
+
+// DefaultRegistryHandler is the struct that implements the RegistryHandler methods
+type DefaultRegistryHandler struct {
+	keychain         authn.Keychain
+	insecureRegistry []string
+}
+
+// NewRegistryHandler creates a new DefaultRegistryHandler
+func NewRegistryHandler(keychain authn.Keychain, insecureRegistries []string) *DefaultRegistryHandler {
+	return &DefaultRegistryHandler{
+		keychain:         keychain,
+		insecureRegistry: insecureRegistries,
+	}
+}
+
+// EnsureReadAccess ensures that we can read from the registry
+func (rv *DefaultRegistryHandler) EnsureReadAccess(imageRefs ...string) error {
+	for _, imageRef := range imageRefs {
+		if err := verifyReadAccess(imageRef, rv.keychain, rv.GetInsecureRegistryOptions(imageRef)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// EnsureWriteAccess ensures that we can write to the registry
+func (rv *DefaultRegistryHandler) EnsureWriteAccess(imageRefs ...string) error {
+	for _, imageRef := range imageRefs {
+		if err := verifyReadWriteAccess(imageRef, rv.keychain, rv.GetInsecureRegistryOptions(imageRef)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GetInsecureRegistryOptions returns a list of WithRegistrySetting imageOptions matching the specified imageRef prefix
+func (rv *DefaultRegistryHandler) GetInsecureRegistryOptions(imageRef string) []remote.ImageOption {
```

Comment:
nit: since this is already a "registry" handler maybe this could just be `GetInsecureOptions`

Reply:
I like it!